tests/test_pydantic_factory.py

```diff
@@ -207,6 +207,23 @@ class FooFactory(ModelFactory[Foo]):
     assert foo.default_field == 10
 
 
+def test_factory_nested_model_collection_coverage() -> None:
```

Comment:
Can we also have a test with a `factory_use_construct=True` 🙏🙏

something like

```python

 def test_factory_nested_model_collection_coverage() -> None:
    class Nested(BaseModel):
        foo: int

        @validator("foo")
        @classmethod
        def always_invalid(cls, v: int) -> None:
            raise ValueError("invalid by validator")

    class CollectionModel(BaseModel):
        collection: List[Nested]

    class CollectionModelFactory(ModelFactory[CollectionModel]):
        __model__ = CollectionModel

    instances = list(CollectionModelFactory.coverage(factory_use_construct=True))
    assert len(instances) == 1
    instance = instances[0]
    assert len(instance.collection) == 1
    assert isinstance(instance.collection[0], Nested)
```

Reply:
I've added this tests (and its passing!)